modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaConnectionProviderModelParser.java

```diff
@@ -102,7 +102,7 @@ public JavaConnectionProviderModelParser(JavaExtensionModelParser extensionModel
         .orElse(ExtensionModel.class.getClassLoader());
 
     collectAdditionalModelProperties();
-    if (mustResolveMinMuleVersion()) {
+    if (loadingContext.isResolveMinMuleVersion()) {
```

Comment:
rather than repeating this on every single parser (for every single supported language), why not instead add a `MuleVersion calculateMinMuleVersion()` method, and have the loader delegates only invoke it when necessary? In that way the amount of repeated code in the parsers decreases, we avoid the possibility of a  parser wrongfully calculating anyway and the concern gets centralized